scripts/python/compute_memory_footprint.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+# NOTE: Currently assumes GRMHD, but not radiation
+# We need to modify this to account for radiation at
+# some point in the future.
+
+import numpy as np
+from argparse import ArgumentParser
+
+def compute_footprint(nb,nx1,nx2,nx3):
+    """Compute footrprint for Phoebus given a number nb of meshblocks,
+    each of size nx1 * nx2 * nx3
+    Output size is in GB. Assumes some set of variables
+    and about a 15% overhead due to unaccounted for memory.
+    """
+    GB2B = (1024*1024*1024)
+    B2GB = 1./GB2B
+
+    ng = 4
+    bytes_per_num = 8
+    nums_per_3d_var = (nx1+2*ng)*(nx2+2*ng)*(nx3+2*ng)
+    nums_per_2d_var = (nx1+2*ng)*(nx2+2*ng)
+    stages = 4 # base, rhs, k, krhs
+    nfaces = 4 # 3 faces + coarse
+    
+    # comm buffers
+    nums_per_ghost_halo = nums_per_3d_var - nx1*nx2*nx3
+    
+    # c.energy, c.bfield, c.density, c.momentum
+    tot_indep_vars =(1 + 3 + 1 + 3)
+    indep_3d_nums = tot_indep_vars*(nums_per_3d_var*stages + nums_per_3d_var*nfaces + nums_per_ghost_halo)
+    # p.energy, cell_signal_speed, face_signal_speed, gamma1, p.bfield, ql, qr, temperature, emf, divb, p.velocity, c2p_mu, c2p_scratch, fail, pressure, p.density, node coords
+    # ql and qr are nindep*ndim
+    dep_3d_nums = (1 + 1 + 3 + 3 + 3 + tot_indep_vars*(3 + 3) + 1 + 3 + 1 + 3 +1 + 1 + 1 + 1 + 1 + 4)*nums_per_3d_var
+    # (centers + faces)*(alpha, dalpha, bcon, gcov, gamcon, detgam, dg, coords)
+    dep_2d_nums = 4*(1 + 4 + 3 + 10 + 6 + 1 + 40 + 4)*nums_per_2d_var
```

Comment:
Not sure if it is worth the time, but could make this into a class that you can register different types of variables with(i.e. `mem_size = memory_footprint(); mem_size.AddIndependent(3, "c.bfield"); mem_size.AddDependent(1, "p.energy");`) and have `compute_footprint` be a class method.

Reply:
This is a nice idea. Probably worth doing when we generalize to radiation, etc. For now though let's leave it. I'll add a TODO in the script.